src/main/java/com/sda/eventapp/web/mvc/controller/EventController.java

```diff
@@ -80,7 +80,7 @@ public String update(ModelMap model, @PathVariable Long id) {
         model.addAttribute("eventImage", eventToUpdate.getImage());
 
         User loggedUser = (User) authenticationFacade.getAuthentication().getPrincipal();
-        if (loggedUser.getId() != (eventService.findOwnerIdByEventId(id))) {
+        if (loggedUser.getId() != (eventService.findOwnerIdByEventId(id)) && !loggedUser.getRole().equals(User.Roles.ROLE_ADMIN)) {
```

Comment:
There is a logic error. Should be `||` instead of `&&`.
Right now we'll get inside of this if only when logged user is an owner. The other check will be skipped.

Reply:
There is no logic error. The && operator is correct. 
The first condition checks if the id of the logged in user **is not** equal to the id of the event owner.
 And the second condition checks if the logged in user **has no** admin role.
 If both are true it will throw 403
 
 I can changed "!loggedUser.getRole().equals(User.Roles.ROLE_ADMIN)" to "loggedUser.getRole() != User.Roles.ROLE_ADMIN" (same but looks better). What do you think?

Reply:
Yup, I think you're right!
I don't think there is need to change anything else here

Reply:
Actually... You can just remove these not necessary parentheses.